tests/test_task.py

```diff
@@ -267,19 +266,32 @@ def test_build_trials_seq_shuffling():
     _ = task_shuffled.generate_trials(ntrials=NTRIALS)
     _ = task_not_shuffled.generate_trials(ntrials=NTRIALS)
 
-    # Verify that the generated sequences are shuffled or not shuffled accordingly
     assert task_shuffled._modality_seq.shape == task_not_shuffled._modality_seq.shape
-    assert not np.array_equal(task_shuffled._modality_seq, task_not_shuffled._modality_seq)
+
+    keys = list(task_shuffled._session.keys())
+    first_occurrence = list(task_not_shuffled._modality_seq).index(keys[1])
+    assert (
+        keys[0] not in task_not_shuffled._modality_seq[first_occurrence:]
+    ), "unshuffled _modality_seq appears shuffled"
+
+    assert not np.array_equal(
+        task_shuffled._modality_seq,
+        task_not_shuffled._modality_seq,
+    ), "shuffled _modality_seq == unshuffled _modality_seq"
 
 
 def test_build_trials_seq_maximum_sequential_trials():
     # Create a Task instance with shuffling enabled and a maximum sequential trial constraint
     task = Task(name=NAME, max_sequential=4)
     _ = task.generate_trials(ntrials=NTRIALS)
+
     # Ensure that no more than the specified maximum number of consecutive trials of the same modality occur
-    for modality in task._modalities:
-        for i in range(len(task._modality_seq) - task.max_sequential):
-            assert np.sum(task._modality_seq[i : i + task.max_sequential] == modality) <= task.max_sequential
+    sequence_string = "".join(task._modality_seq).replace("catch", "X")
```

Comment:
Why is the replace needed?

Reply:
Otherwise `set(sequence_string)` would add each letter of "catch" as a separate test case. There's probably other ways around this, but this seemed simple and sensible to me.

In fact, I was thinking whether we want to replace "catch" by "X" in session (and everywhere) as well. I think given that we say that each modularity is given by a single character, it makes sense to apply this to catches as well, and reserve X (or underscore?) for that purpose.

Reply:
Otherwise `set(sequence_string)` would add each letter of "catch" as a separate test case. There's probably other ways around this, but this seemed simple and sensible to me.

In fact, I was thinking whether we want to replace "catch" by "X" in session (and everywhere) as well. I think given that we say that each modularity is given by a single character, it makes sense to apply this to catches as well, and reserve X (or underscore?) for that purpose.

Reply:
I like this! Let's change it everywhere in the code. I prefer `X` to the underscore. I am in favor of doing it in this PR, but in case you prefer feel free to open a new issue for this.

Reply:
Done